exercises/src/main/scala/fpinscala/datastructures/List.scala

```diff
@@ -49,20 +51,224 @@ object List { // `List` companion object. Contains functions for creating and wo
   def product2(ns: List[Double]) =
     foldRight(ns, 1.0)(_ * _) // `_ * _` is more concise notation for `(x,y) => x * y`; see sidebar
 
+  def tail[A](l: List[A]): List[A] = l match {
+    //case Nil => sys.error("tail of Nil list")
```

Comment:
If you want to follow the 'canonical' implementation of `List.tail` in the Scala collections library, the tail of an empty list should be an error. So the commented-out code is probably more correct.